app/uk/gov/hmrc/selfservicetimetopay/controllers/ArrangementController.scala

```diff
@@ -57,13 +57,15 @@ class ArrangementController(ddConnector: DirectDebitConnector,
       authorizedForSsttp {
         val sa = authContext.principal.accounts.sa.get
 
-        taxPayerConnector.getTaxPayer(sa.utr.utr).flatMap {
-          _.fold(throw new RuntimeException("No taxpayer found"))(t => {
+        taxPayerConnector.getTaxPayer(sa.utr.utr).flatMap[Result] {
+          _.fold(Future.successful(Redirect(routes.SelfServiceTimeToPayController.getTtpCallUs())))(t => {
             updateOrCreateInCache(found => found.copy(taxpayer = Some(t)),
               () => TTPSubmission(taxpayer = Some(t)))
               .map(cm => cm.getEntry("ttpSubmission")(submissionFormatter).get).flatMap[Result] {
-              case TTPSubmission(_, _, _, Some(Taxpayer(_, _, Some(tpSA))), _, _, CalculatorInput(empty@Seq(), _, _, _, _, _)) =>
-                Future.successful(Redirect(routes.CalculatorController.getCalculateInstalments(None)))
+              case ttpData @ TTPSubmission(_, _, _, Some(Taxpayer(_, _, Some(tpSA))), _, _, CalculatorInput(empty@Seq(), _, _, _, _, _)) =>
+                sessionCache.put(ttpData.copy(calculatorData = CalculatorInput(startDate = LocalDate.now(),
```

Comment:
At a brief glance, this seems to violate the 1 get, 1 put rule on keystore interactions? The outer updateOrCreate and then this put.